packages/flutter/lib/src/material/carousel.dart

```diff
@@ -833,6 +877,10 @@ class _RenderSliverFixedExtentCarousel extends RenderSliverFixedExtentBoxAdaptor
       return 0;
     }
 
+    if (infinite) {
+      return (scrollOffset / maxExtent).floor();
+    }
+
     final int firstVisibleIndex = (constraints.scrollOffset / maxExtent).floor();
```

Comment:
I think we should just use scrollOffset here instead of constraints.scrollOffset.